src/Discovery/Walk.hs

```diff
@@ -1,46 +1,53 @@
 module Discovery.Walk
   ( -- * Walking the filetree
-    walk
-  , WalkStep(..)
-
-  , dirName
-  , fileName
-  ) where
-
-import Prologue
+    walk,
+    WalkStep (..),
+    fileName,
+  )
+where
 
+import Control.Monad.IO.Class (MonadIO)
+import Data.Maybe (mapMaybe)
+import Data.Text (Text)
+import qualified Data.Text as T
+import Path
 import Path.IO
+import Prelude
 
 data WalkStep
-  = WalkContinue -- ^ Continue walking subdirectories
-  | WalkSkipSome [Path Abs Dir] -- ^ Skip some subdirectories
-  | WalkSkipAll -- ^ Skip all subdirectories
-  | WalkStop -- ^ Stop walking the filetree entirely
-  deriving (Eq, Ord, Show, Generic)
+  = -- | Continue walking subdirectories
+    WalkContinue
+  | -- | Skip some subdirectories
+    WalkSkipSome [Text]
+  | -- | Skip all subdirectories
+    WalkSkipAll
+  | -- | Stop walking the filetree entirely
+    WalkStop
+  deriving (Eq, Ord, Show)
 
 -- | Walk the filetree, rooted at an absolute directory. The passed-in function
 -- takes the @currentdir@ @subdirs@ and @files@, and produces a WalkStep
 -- describing what to do next.
 --
 -- You can inspect the names of files and directories with 'dirName' and
 -- 'fileName'
-walk
-  :: MonadIO m
-  => (Path Abs Dir -> [Path Abs Dir] -> [Path Abs File] -> m WalkStep)
-  -> Path Abs Dir
-  -> m ()
+walk ::
+  MonadIO m =>
+  (Path Abs Dir -> [Path Abs Dir] -> [Path Abs File] -> m WalkStep) ->
+  Path Abs Dir ->
+  m ()
 walk f = walkDir $ \dir subdirs files -> do
   -- normally, subdirs and files are _relative to dir_. We want them to be
   -- relative to the filetree walk
   step <- f dir subdirs files
   case step of
-    WalkContinue -> pure (WalkExclude [])
-    WalkSkipSome dirs -> pure (WalkExclude dirs)
-    WalkSkipAll -> pure (WalkExclude subdirs)
+    WalkContinue -> pure $ WalkExclude []
+    WalkSkipSome dirs ->
```

Comment:
Does this solve the `WalkSkipSome` compilation errors you mentioned?

Reply:
It does -- I'm genuinely surprised there weren't compilation errors before